wsproto/frame_protocol.py

```diff
@@ -142,6 +142,28 @@ def _consume_exactly(self, nbytes):
             yield
         return (yield from self._consume_at_most(nbytes))
 
+    def _parse_extended_payload_length(self, opcode, payload_len):
+        if opcode.iscontrol() and payload_len > 125:
+            raise ParseFailed("Control frame with payload len > 125")
+        if payload_len == 126:
+            data = yield from self._consume_exactly(2)
+            (payload_len,) = struct.unpack("!H", data)
+            if payload_len <= 125:
+                raise ParseFailed(
+                    "Payload length used 2 bytes when 1 would have sufficed")
+        elif payload_len == 127:
+            data = yield from self._consume_exactly(8)
+            (payload_len,) = struct.unpack("!Q", data)
+            if payload_len < 2 ** 16:
```

Comment:
So this is in section 5.2 of RFC 6455. The payload length can be expressed in three different ways and it's required that you use the one most appropriate. I agree that we should be using names here for comprehensibility purposes but the standard doesn't give us any obvious names. The values in question are:

- 125 (Maximum normal (or short?) length)
- 126 (Flag value for two-byte extended length)
- 127 (Flag value for eight-byte extended length)
- 2^16 (Maximum two-byte extended length)
- 2^64 (Maximum eight-byte extended length)

I don't think the last one is used but having names for these would be handy. If I were to suggest some I'd probably go with:

```
PAYLOAD_LENGTH_TWO_BYTE = 126
PAYLOAD_LENGTH_EIGHT_BYTE = 127
MAX_PAYLOAD_LENGTH_SHORT = 125
MAX_PAYLOAD_LENGTH_TWO_BYTE = 2 ** 16
MAX_PAYLOAD_LENGTH_EIGHT_BYTE = 2 ** 64
```

I'm open to discussion though.